In [1]:
from math import factorial

# Задача 1

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, 
если в этот день должно быть 4 разных занятия?

Решение :
в пн 4 разных занятия из 10 занятий, значит используем формулу для размещений,тут важен порядок

In [2]:
def arr(n, k):
    return (factorial(n) / factorial(n - k))

In [3]:
arr(10, 4)

5040.0

Ответ: 5040 способов

# Задача 2

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Решение: всего в колоде 4 туза, значит вероятность вытянуть хотя бы 1 туза - 4 к 52

In [4]:
ver_tuz=(4/52)
ver_tuz

0.07692307692307693

найдм сколько всего комбинации и умножим на вероятность достать туза

In [5]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

порядок не важен, используем формулу для сочетаний

In [6]:
all_variannts=combinations(52,4)
all_variannts

270725

In [7]:
comb_tuz=all_variannts*ver_tuz
print(comb_tuz)

20825.0


ответ : 20825 исходов

# Задача 4

Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. 
Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?